### PR TITLE
android buildfix

### PR DESCRIPTION
**Pull request by grant2258**

### CODE REVIEW

File: src/osd/libretro/retromain.cpp
Comment (roslof) [inline src/osd/libretro/retromain.cpp:1]: Accidental space added to line 201?

Comment (grant2258) [reply src/osd/libretro/retromain.cpp:1]: already fixed before your comment see  https://github.com/libretro/mame/pull/383/commits/4c6daefc0d42b7ddf5aa3b9fa676e7b1e91f588c good spot though

**Comment by grant2258** [comment]
@sonninnos  try build with this makefile if you get no joy ill add debugging to see whats going on with the gcc detection. Went through the lua and makefile all you need set for mame is the ndk_home. Can you post any errors if there is any would be helpful

```
 make -f Makefile.libretro platform=android-arm64 SUBTARGET=tiny -j7
```
or

```
make OSD=retro CONFIG=libretro  android-arm64 VERBOSE=1 SUBTARGET=tiny  -j7 REGINIE=1 
```
anyone is fine not sure if the makefile is setting the correct homepath if its different just set it.

ANDROID_NDK_ROOT and other old variables where changed to one single variable ANDROID_NDK_HOME from master too 255 branch thats what cause the failure. Hope that helps clarify.

git difftool master..libretro/mame-0.255 --  scripts/toolchain.lua

**Comment by sonninnos** [comment]
Thanks, let's see what buildbot says..

**Comment by sonninnos** [comment]
Cool, now we are in the next territory: https://git.libretro.com/libretro/mame/-/pipelines/204224

```
/builds/libretro/mame/scripts/genie.lua:590: Definition file for TARGET=mame SUBTARGET=arcade does not exist
stack traceback:
	[C]: in function 'error'
	/builds/libretro/mame/scripts/genie.lua:590: in main chunk
	[C]: in upvalue 'builtin_dofile'
	[string "premake = { }..."]:109: in function 'dofile'
	[string "_WORKING_DIR        = os.getcwd()..."]:46: in function '_premake_main'
make[1]: *** [makefile:1286: build/projects/retro/mamearcade/gmake-android-x64/Makefile] Error 1
```

So either we need to bring back that subtarget stuff, or just not use them in android builds..


Edit: The old `arcade.flt` required just a tiny bit of editing. Let's add the old `mess.flt` too then.

**Comment by sonninnos** [comment]
Ok, now all android builds are back to stopping at:
```
3rdparty/genie/bin/linux/genie --distro=generic --CC='clang' --OPTIMIZE=3 --NOWERROR='1' --target='mame' --subtarget='arcade' --build-dir='build' --NO_USE_MIDI='1' --NO_USE_PORTAUDIO='1' --PYTHON_EXECUTABLE='python3' --osd='retro' --targetos='linux' --PLATFORM='unknown' --gcc=android-arm64 --gcc_version= --osd=retro --targetos=android --PLATFORM=arm64 --NO_USE_MIDI=1 --NO_OPENGL=1 --USE_QTDEBUG=0 --DONT_USE_NETWORK=1 --NOASM=1 gmake
1549 source file(s) found
Clang version 6.0 or later needed
make[1]: *** [makefile:1242: build/projects/retro/mamearcade/gmake-android-arm64/Makefile] Error 255
```

**Comment by grant2258** [comment]
are you specifically testing setting the targets  variable because them subtragets where done away with. As for your compile error something is setting targetos to linux then later its set to android. I would need to see how your building it to move forward with the specific area

**Comment by grant2258** [comment]
The arcade and mess builds are not an issue unless you specifically are setting them somewhere do you have the makefile call and the full log to the latest fail?

**Comment by sonninnos** [comment]
The failure logs are here: https://git.libretro.com/libretro/mame/-/pipelines/204225

`.gitlab-ci.yml` has the `SUBTARGET` stuff, as in how everything is built, and it is using this template: https://git.libretro.com/libretro-infrastructure/ci-templates/-/blob/master/android-make.yml

**Comment by grant2258** [comment]
looks like its using makefile as apposed to makefile.libretro so  ANDROID_NDK_HOME is not set or in the export list(the others are defunct now anyway. 

change{
```
script:
    - make -C ${MAKEFILE_PATH} -f ${MAKEFILE} -j${NUMPROC} platform=${ANDROID_PLATFORM}
```
to 
```
 - make -C ${MAKEFILE_PATH} -f ${MAKEFILE} -j${NUMPROC} ${ANDROID_PLATFORM} ANDROID_NDK_HOME=/opt/ndk OSD=retro CONFIG=libretro
```


also add export ANDROID_PLATFORM ANDROID_NDK_HOME to the export list


im not sure where the ndk is located though is assume the makefile.libretro /opt/ndk is correct

I also see no arcade definition anywhere so that must be coming from somewhere else as well


edit found teh subtargets in the mame repo

https://github.com/libretro/mame/blob/master/.gitlab-ci.yml 

the arcade and mess are set there so you can use them or not use them the question is. Now I can see a little more if the buildbot forget the above for now a let me try a fix I can commit

**Comment by grant2258** [comment]
Ive put a pull request in this should export the variable we need as its not finding the compilers. If there is still any issues ill add some debug stuff in to see if the nkk path is vaild

**Pull request merged**